packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -342,10 +360,30 @@ export function buildExtraViteConfig(
 				plugins: [{ name: facadeEsbuildSveltePluginName, setup: () => {} }]
 			}
 		};
+	} else {
+		if (options.disableDependencyReinclusion !== true) {
+			const disabledReinclusions = options.disableDependencyReinclusion || [];
+			if (disabledReinclusions.length > 0) {
+				log.debug(`not reincluding transitive dependencies of`, disabledReinclusions);
+			}
+			const transitiveDepsToInclude = depsConfig.optimizeDeps.include.filter((dep) =>
+				disabledReinclusions.every((disabled) => !dep.includes(disabled))
+			);
+			log.debug(
+				`reincluding transitive dependencies of excluded svelte dependencies`,
+				transitiveDepsToInclude
+			);
+			depsConfig.optimizeDeps.include?.push(...transitiveDepsToInclude);
+		}
+		extraViteConfig.optimizeDeps.include?.push(...depsConfig.optimizeDeps.include);
+		extraViteConfig.optimizeDeps.exclude?.push(...depsConfig.optimizeDeps.exclude);
 	}
 
-	// @ts-ignore
-	extraViteConfig.ssr = buildSSROptionsForSvelte(svelteDeps, options, config, extraViteConfig);
+	// ssr config
+	extraViteConfig.ssr = buildSSROptionsForSvelte(config);
+	// @ts-expect-error `noExternal` is 100% an array
+	extraViteConfig.ssr?.noExternal?.push(...depsConfig.ssr.noExternal);
```

Comment:
the push only happens when extraViteConfin.ssr exists, otherwise depsConfig values will be ignored. Is that correct?

Reply:
`buildSSROptionsForSvelte` would ensure `ssr.noExternal` always exist, but I can probably refactor this to make the intent clearer.

Reply:
if it is always set, the `?` operator should not be there.